servicedirectory/servicedirectory.rb

```diff
@@ -0,0 +1,269 @@
+  # Initialize the client
+  client = Google::Cloud::ServiceDirectory.registration_service
+
+  # The path of the service
+  service_path = client.service_path(
+    project:   project,
+    location:  location,
+    namespace: namespace,
+    service:   service
+  )
+
+  # Use the Service Directory API to delete the service
+  client.delete_service name: service_path
+  puts "Deleted service: #{service_path}"
+  # [END servicedirectory_delete_service]
+end
+
+def create_endpoint project:, location:, namespace:, service:, endpoint:
+  # [START servicedirectory_create_endpoint]
+  # project   = "Your Google Cloud project ID"
+  # location  = "The Google Cloud region containing the namespace"
+  # namespace = "The name of the parent namespace"
+  # service   = "The name of the parent service"
+  # endpoint  = "The name of the endpoint you are creating"
+
+  require "google/cloud/service_directory"
+
+  # Initialize the client
+  client = Google::Cloud::ServiceDirectory.registration_service
+
+  # The parent path of the endpoint
+  parent = client.service_path(
+    project: project,
+    location: location,
+    namespace: namespace,
+    service: service
+  )
+
+  # Set the IP Address and Port on the Endpoint
+  endpoint_data = Google::Cloud::ServiceDirectory::V1beta1::Endpoint.new(
+    address: "10.0.0.1",
+    port:    443
+  )
+
+  # Use the Service Directory API to create the endpoint
+  response = client.create_endpoint(
+    parent: parent, endpoint_id: endpoint, endpoint: endpoint_data)
+  puts "Created endpoint: #{response.name}"
+  # [END servicedirectory_create_endpoint]
+end
+
+def delete_endpoint project:, location:, namespace:, service:, endpoint:
+  # [START servicedirectory_delete_endpoint]
+  # project   = "Your Google Cloud project ID"
+  # location  = "The Google Cloud region containing the namespace"
+  # namespace = "The name of the parent namespace"
+  # service   = "The name of the parent service"
+  # endpoint  = "The name of the endpoint"
+
+  require "google/cloud/service_directory"
+
+  # Initialize the client
+  client = Google::Cloud::ServiceDirectory.registration_service
+
+  # The path of the endpoint
+  endpoint_path = client.endpoint_path(
+    project:   project,
+    location:  location,
+    namespace: namespace,
+    service:   service,
+    endpoint:  endpoint
+  )
+
+  # Use the Service Directory API to delete the endpoint
+  client.delete_endpoint name: endpoint_path
+  puts "Deleted endpoint: #{endpoint_path}"
+  # [END servicedirectory_delete_endpoint]
+end
+
+def resolve_service project:, location:, namespace:, service:
+  # [START servicedirectory_resolve_service]
+  # project   = "Your Google Cloud project ID"
+  # location  = "The Google Cloud region containing the namespace"
+  # namespace = "The name of the parent namespace"
+  # service   = "The name of the service"
+
+  require "google/cloud/service_directory"
+
+  # Initialize the client
+  client = Google::Cloud::ServiceDirectory.lookup_service
```

Comment:
super nit, I prefer slightly more descriptive names, esp as ServiceDirectory has multiple factory methods for clients. (i.e. lookup_service = Goo......)

Reply:
Done